pkg/adapter/quotas.go

```diff
@@ -103,4 +107,16 @@ type (
 		// The total amount of quota returned, may be less than requested.
 		Amount int64
 	}
+
+	// QuotaResult2 provides return values from quota allocation calls on the handler
+	QuotaResult2 struct {
+		// The outcome status of the operation.
+		Status rpc.Status
```

Comment:
I don't think we need status for Quota. I think it's enough to have:

-error != nil to indicate there was an error and the quota operation didn't execute.
-error == nil to indicate the quota operation executed, Amount indicates the amount allocated which can be 0.

Status is not necessary.